providers/resource/bin/WIFI_phy.py

```diff
@@ -0,0 +1,134 @@
+    bands_data = output.split("Supported commands:")[0]
+    bands = {}
+    for band in bands_data.split("Band "):
+        if band:
+            band_ptn = r"(\d+):\s+([\s\S]*)"
+            band_compile = re.compile(band_ptn)
+            band_res = band_compile.match(band)
+            if band_res:
+                band_num = band_res.groups()[0]
+                band_content = band_res.groups()[1]
+                freqs_raw = band_content.split("Frequencies:")[1]
+                freq_ptn = r"(\d+ MHz.*)"
+                freq_compile = re.compile(freq_ptn)
+                freqs = [
+                    freq
+                    for freq in freq_compile.findall(freqs_raw)
+                    if "disabled" not in freq
+                ]
+                bands[band_num] = freqs
+    return bands
+
+
+def check_sta_support(phy_info_output: str) -> Dict[str, str]:
+    """
+    Checks if supported STAs (BE, AX, AC) are present based on keywords in
+    the output.
+    """
+    # Supported STAs and their keywords
+    supported_stas = {"BE": "EHT", "AX": "HE RX MCS", "AC": "VHT RX MCS"}
+
+    sta_supported = {
+        sta: "supported" if sta_keyword in phy_info_output else "unsupported"
+        for sta, sta_keyword in supported_stas.items()
+    }
+    return sta_supported
+
+
+def check_freq_support(bands: Dict[str, List[str]]) -> Dict[str, str]:
+    """
+    Checks if supported frequency (2.4GHz, 5GHz, 6GHz) are present based on
+    band number
+    """
+    supported_freqs = {"2.4GHz": "1", "5GHz": "2", "6GHz": "4"}
+
+    freq_supported = {
+        freq: (
+            "supported"
+            if band in bands and len(bands[band]) > 0
+            else "unsupported"
+        )
+        for freq, band in supported_freqs.items()
+    }
+    return freq_supported
+
+
+def create_phy_interface_mapping(phy_interface: List[Tuple[str, str]]) -> Dict:
+    """
+    Creates a mapping between interfaces and their PHY, bands, and STA support.
+    """
+    phy_interface_mapping = {}
+    for phy, interface in phy_interface:
+        cmd = "iw {} info".format(phy)
+        phy_info_output = check_output(
+            cmd, shell=True, universal_newlines=True
+        )
+        bands = parse_phy_info_output(phy_info_output)
+        freq_supported = check_freq_support(bands)
+        sta_supported = check_sta_support(phy_info_output)
+        phy_interface_mapping[interface] = {
+            "PHY": phy,
+            "Bands": bands,
+            "FREQ_Supported": freq_supported,
+            "STA_Supported": sta_supported,
+        }
+    return phy_interface_mapping
+
+
+def main():
+    # Read and parse "iw dev" output
+    phy_interface = parse_iw_dev_output()
+
+    # Create mapping with interface, PHY, bands, and supported STAs
+    phy_interface_mapping = create_phy_interface_mapping(phy_interface)
+
+    # Print interface summary with detailed information on separate lines
+    for interface, content in phy_interface_mapping.items():
```

Comment:
**Major issue:** This resource has a problem. If you have more than 1 wifi interface they are all going to be printed one after the other, this makes them all "the same" resource, when in fact they refer to different things. This makes it so to refer to any of the fields, you are going to need to know the name of the interfaces because it is baked into the field name.
Sample output:
```
wlan0_2_4GHz: supported
wlan0_5GHz: supported
wlan0_6GHz: unsupported
wlan0_be: unsupported
wlan0_ax: supported
wlan0_ac: supported
wlan1_2_4GHz: supported
wlan1_5GHz: supported
wlan1_6GHz: unsupported
wlan1_be: unsupported
wlan1_ax: supported
wlan1_ac: supported
```

I think this is not what you want at all, as it makes this basically pointless without having another resource that lists all wireless interfaces, and even at that you need a jinja template just to use this. (I see this is exaclty what you used: https://github.com/canonical/checkbox/pull/1224/files#diff-84452daf18e36a46b455951936c8dd286eece5cd8474cee63033e35d74c35cdaR287

If this was in the form of
```
name: wlan0
2_4GHz: supported
5GHz: supported
6GHz: unsupported
be: unsupported
ax: supported
ac: supported

name: wlan1
2_4GHz: supported
5GHz: supported
6GHz: unsupported
be: unsupported
ax: supported
ac: supported
```
You could use this resource directly to expand your template or query it in a more intuitive way. [For example for this one](https://github.com/canonical/checkbox/pull/1224/files#diff-84452daf18e36a46b455951936c8dd286eece5cd8474cee63033e35d74c35cdaR267)

This doesn't do exactly the same (namely, if device contains an interface that iw doesn't see, this will not create a skipped job as your version does)
```
unit: template
template-resource: wireless_sta_protocol
template-filter: wireless_sta_protocol.be == 'supported'
template-engine: jinja2
template-unit: job
id: wireless/wireless_connection_wpa_be_nm_{{ interface }}
template-id: wireless/wireless_connection_wpa_be_nm_interface
depends: wireless/detect
_summary: Connect to WPA-encrypted 802.11be Wi-Fi network on {{ interface }}
_purpose:
  Check system can connect to 802.11be AP with wpa security
plugin: shell
user: root
command:
  net_driver_info.py "$NET_DRIVER_INFO"
  wifi_nmcli_test.py secured {{ interface }} "$WPA_BE_SSID" "$WPA_BE_PSK"
category_id: com.canonical.plainbox::wireless
estimated_duration: 30.0
flags: preserve-locale also-after-suspend
```

This is exactly the same but it is easier to read in my opinion (note the `(` `)` around the expression, you must use them in this case)
```
unit: template
template-resource: device
template-filter: device.category == 'WIRELESS' and device.interface != 'UNKNOWN'
template-engine: jinja2
template-unit: job
id: wireless/wireless_connection_wpa_be_nm_{{ interface }}
template-id: wireless/wireless_connection_wpa_be_nm_interface
depends: wireless/detect
_summary: Connect to WPA-encrypted 802.11be Wi-Fi network on {{ interface }}
_purpose:
  Check system can connect to 802.11be AP with wpa security
plugin: shell
user: root
command:
  net_driver_info.py "$NET_DRIVER_INFO"
  wifi_nmcli_test.py secured {{ interface }} "$WPA_BE_SSID" "$WPA_BE_PSK"
category_id: com.canonical.plainbox::wireless
estimated_duration: 30.0
flags: preserve-locale also-after-suspend
requires:
 (wireless_sta_protocol.name == '{{ interface }}' and wireless_sta_protocol.be == 'supported')
```

Reply:
For now, I will focus on refactoring the resource job only.
I know what you're saying, and this thing is in my backlog.
If want to do this, it will impact all Wireless test job, and I want to avoid in this PR.
I will implent all Wireless test job to use the same RESONALBE template-resource.
It will also need to merge the "net_if_managment" resource job.

I would like to not using the template-filter to not generate the job, and will make the template job will generate the correct job. Those job not support (like not support BE) should auto skip not filter out.

The followings are what I plan to do:
1. Not using the device as the template-resource
2. Merge two resource job (wireless_sta_protocol, net_if_managment) into one, and utillize it as all Wireless test job's template-resource.

Reply:
ok fantastic, so this will be a follow up